src/Provider/Nominatim/Nominatim.php

```diff
@@ -54,7 +54,7 @@
      *
      * @return Nominatim
      */
-    public static function withOpenStreetMapServer(HttpClient $client, string $userAgent, string $referer = '')
+    public static function withOpenStreetMapServer(HttpClient $client, string $userAgent, string $referer = ''): self
```

Comment:
I think adding this is a BC break. Or is this class `final`?

Reply:
I think this class could be declared as `final` !
But I can remove the `self`, it doesn't "add" a lot.

Reply:
I just checked : the class is (and was) indeed `final` : https://github.com/geocoder-php/Geocoder/blob/master/src/Provider/Nominatim/Nominatim.php#L33